src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,19 +106,22 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
-        } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
         } catch (JAXBException jaxbe) {
             throw new StorageOperationException("Error converting address book into storage format");
-        }
+        } catch (FileNotFoundException fnfe) {
```

Comment:
You can discuss with me on this activity :-) as there may be some misunderstanding of how and why the exception is needed/used in this case.

In this method, you should detect the file missing and _throws_ an exception.

The catch (JAXBException) above is to rethrow an exception (basically change the exception into another type of exception) so that the outsiders are not aware of specific implementation details (e.g. JAXB is a specific way to save object into files, but we dont want the outsider to know this, so we rethrow JAXB specific exception into a generic StorageOperationException).

For our file not found case, you can just throw the exception from this method without the same catch-and-rethrow. :-)